modules/blake3/1.3.3/patches/add_build_file.patch

```diff
@@ -0,0 +1,65 @@
+--- /dev/null
++++ BUILD.bazel
+@@ -0,0 +1,62 @@
++load("@rules_cc//cc:defs.bzl", "cc_binary", "cc_library")
++
++cc_library(
++    name = "blake3",
++    srcs = [
++        "c/blake3.c",
++        "c/blake3_dispatch.c",
++        "c/blake3_portable.c",
++    ] + select({
++        "@bazel_tools//src/conditions:linux_x86_64": [
++            "c/blake3_avx2_x86-64_unix.S",
++            "c/blake3_avx512_x86-64_unix.S",
++            "c/blake3_sse2_x86-64_unix.S",
++            "c/blake3_sse41_x86-64_unix.S",
++        ],
++        "@bazel_tools//src/conditions:windows_x64": [
++            "c/blake3_avx2_x86-64_windows_msvc.asm",
++            "c/blake3_avx512_x86-64_windows_msvc.asm",
++            "c/blake3_sse2_x86-64_windows_msvc.asm",
++            "c/blake3_sse41_x86-64_windows_msvc.asm",
++        ],
++        "@bazel_tools//src/conditions:darwin_arm64": [
++            "c/blake3_neon.c",
++        ],
++        "//conditions:default": [],
++    }),
++    hdrs = [
++        "c/blake3.h",
++        "c/blake3_impl.h",
++    ],
++    copts = select({
++        "@bazel_tools//src/conditions:linux_x86_64": [],
++        "@bazel_tools//src/conditions:windows_x64": [],
++        "@bazel_tools//src/conditions:darwin_arm64": [
++            "-DBLAKE3_USE_NEON=1",
++        ],
++        "//conditions:default": [
++            "-DBLAKE3_NO_SSE2",
++            "-DBLAKE3_NO_SSE41",
++            "-DBLAKE3_NO_AVX2",
++            "-DBLAKE3_NO_AVX512",
++        ],
++    }),
++    includes = ["."],
++    visibility = ["//visibility:public"],
++)
++
++cc_binary(
++    name = "example",
++    srcs = [
++        "c/example.c",
++    ],
++    copts = [
++        "-w",
++        "-O3",
```

Comment:
This fails on Windows: https://buildkite.com/bazel/bcr-presubmit/builds/1470#0188c2d0-adc3-491c-82b4-10e4eb738f73/250-327

Reply:
The flag does generate a warning, but the failure actually happens because of `unistd.h` imported in `example.c` does not exist on Windows.

Looking at https://github.com/BLAKE3-team/BLAKE3/pull/48/files, it's possible that this dependency could be safely removed.

Reply:
I just removed example.c for now.